src/clipping/weileratherton.jl

```diff
@@ -0,0 +1,146 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    WeilerAtherton()
+
+The Weiler-Atherton algorithm for clipping polygons.
+
+## References
+ 
+* Weiler, Kevin and Atherton, Peter. 1977. [Hidden surface removal using polygon area sorting]
+  (https://dl.acm.org/doi/10.1145/563858.563896)
+"""
+struct WeilerAtherton <: ClippingMethod end
+
+function clip(poly::Polygon, other::Geometry, method::WeilerAtherton)
+  if hasholes(poly)
+    @error "not implemented"
+  end
+
+  c = [clip(ring, boundary(other), method) for ring in rings(poly)]
+  r = filter(!isnothing, vcat(c...))
+
+  if isempty(r)
+    nothing
+  elseif length(r) > 1
+    Multi(PolyArea.(r))
+  else
+    PolyArea(r[1])
+  end
+end
```

Comment:
This method is already available in src/clipping.jl, we only need to add the method with rings in this file.